pkg/report/predicate/vuln.go

```diff
@@ -0,0 +1,91 @@
+package predicate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/package-url/packageurl-go"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/clock"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+// CosignVulnPredicate represents the Cosign Vulnerability predicate.
+// Cosign provides the CosignVulnPredicate structure in their repository.
+// But the type of Scanner.Result is defined as map[string]interface{}, which is difficult to use,
+// so we define our own.
+// The PR is in progress to replace Scanner.Result type to interface{}.
+// https://github.com/sigstore/cosign/pull/2096
```

Comment:
This PR got merged.

Reply:
I have added comments on the background and references.